apps/angular/src/core/services/anime-parameters.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AnimeType } from '@js-camp/core/models/anime';
+import { AnimeParameters } from '@js-camp/core/models/anime-parameters';
+import { AnimeSortingField } from '@js-camp/core/models/anime-sorting-field';
+import { PaginationParameters } from '@js-camp/core/models/pagination-parameters';
+import { SortingDirection, SortingParameters } from '@js-camp/core/models/sorting-parameters';
+import { EnumUtils } from '@js-camp/core/utils/enum.utils';
+import { Observable, map } from 'rxjs';
+
+type AnimePageQueryParams = Partial<Record<keyof AnimeParameters, string | null>>;
```

Comment:
'null' in type is redundant here because we have `Partial` type wrapper.
```suggestion
type AnimePageQueryParams = Partial<Record<keyof AnimeParameters, string>>;
```

Reply:
I know we have to add 'null' to use it with 'SortingParameters'.
However, I suppose we can adjust 'SortingParameters' type instead. It looks like 'null' inside the 'SortingParameters' is redundant. 
For instance, object below doesn't make sense to me:
```ts
const sortParams: SortingParameters = {
  filed: null,
  direction: SortingDirection.Ascending,
}
```

I suppose, either we should have both properties inside 'SortingParameters' or the params itself should be empty.
Let's try to refactor it.